### DISCUSSION TITLE
Default RADI endpoint not working on WSL2

### DISCUSSION THREAD

**Discussion by Bhardwaj-Himanshu** [JdeRobot/RoboticsAcademy#2036] [Q&A] [answered]
**Issue**
![image](https://user-images.githubusercontent.com/74407676/235411045-cf5dcb29-87d4-4b2f-8f89-e6fe8ec5e332.png)

**Steps to reproduce**
Launch an instance of docker container `jderobot/robotics-academy:latest` , the port redirects to https://0.0.0.8000/

**Steps to reproduce**
Launch an instance of docker container `jderobot/robotics-academy:latest` , the port redirects to https://0.0.0:8000/

**Behaviour**
Should show instance deployed at https://0.0.0.0:8000/--> which refuses to connect, instead I need to connect to https://127.0.1:8000/ manually!

**Comment by pawanw17** [comment]
Hi @Bhardwaj-Himanshu can you tell more about the system on which you are seeing this bug? Is it a VM? what kind of connections you have? ubuntu version?

**Reply by Bhardwaj-Himanshu** [reply]
Yup @pawanw17 
So the things in asked order are as follows-`the machine it's not working on`
Ubuntu 22.04 
WSL 2 on Windows 10 Home

Idk what if I understand about connections, could you please eleborate it a bit? Like ports or IP addresses allowed?

Also,tried to reproduce this issue with a Ubuntu Cloud Based instance, but could not do so, So I think this could be a `my-machine` specific issue? But I wanna know, if or how ports work, because currently for me on WSL2 0.0.0.8000-->localhost:3000 on a windows edge browser as WSL2 provides a bridge between ubuntu and windows!

**Reply by pawanw17** [reply]
Correct me if I am wrong,
You are running the RADI in Ubuntu 22.04 (not a VM)? 

This issue is likely specific to your machine.
About ports, Basically, they are used to set up multiple connections to a single IP for example in Robotics Academy the IP 127.0.0.1 has multiple connections for different purposes:
Look here for more: https://github.com/JdeRobot/RoboticsInfraestructure/tree/noetic-devel/scripts#radi-port-list
https://stackoverflow.com/questions/20778771/what-is-the-difference-between-0-0-0-0-127-0-0-1-and-localhost

**Reply by Bhardwaj-Himanshu** [reply]
Yep, it most probably likely to my machine where it redirects me to ports :3000 and 127.0.0.1!

Looks like I need to bingewatch an interesting video/series over youtube, with some popcorn!
Thanks for all the input @pawanw17

**Reply by pawanw17** [reply]
Hi @Bhardwaj-Himanshu I still don't understand the machine you are using is it ubuntu 22 or windows?

**Reply by Bhardwaj-Himanshu** [reply]
Can't say it's ubuntu 22 on windows, either I can run both side by side in a dual boot or install a virtual machine box software or the third way, live integration of Linux on windows [WSL2](https://learn.microsoft.com/en-us/windows/wsl/about), which uses a hypervisor to allot half the resources for both the OS. Like a container with different machines and different operating systems!

**Reply by pawanw17** [reply]
Hi, I am sorry, What I mean to ask is when you are seeing the issue, what is the system you are using, I am not understanding whether you are using ubuntu 22 or WSL2

**Reply by Bhardwaj-Himanshu** [reply]
The issue is observed on WSL2
No issues if I run it on a Ubuntu Machine or Ubuntu Cloud instance!

Also, it's my bad as well for misinterpreting the question.

**Reply by pawanw17** [reply]
Hi Sorry for delay, looking around this seems to be a known issue of wsl2
https://stackoverflow.com/questions/69926941/localhost-refused-to-connect-on-wsl2-when-accessed-via-https-localhost8000-b

https://abdus.dev/posts/fixing-wsl2-localhost-access-issue/

You can try one of the above solutions (there are more available I think) but I suggest just use to 127.0.0.1:8000 for WSL2 since this is more of a WSL2 issue than a Robotics Academy issue.

Robotics Academy uses 0.0.0.0 since that is also the default IP docker uses and also accepts any connections on 127.0.0.1 and if you see, it is hard coded at a lot of places,

![image](https://github.com/JdeRobot/RoboticsAcademy/assets/40732709/02d372d7-723a-4e16-9d88-f529161cd5cd)

![image](https://github.com/JdeRobot/RoboticsAcademy/assets/40732709/ff2b84ca-dcc0-4981-970b-45b145d59ddb)

@pariaspe @jmplaza thoughts?

**Reply by Bhardwaj-Himanshu** [reply]
No worries for the delay mate, no rush!
Also, yup now this clarifies some of my doubts regarding the same and how to get around it.
Thanks for the much-needed help.

@pawanw17 

Also, just a out of curiosity question here, do you use wsl2 as your primary coding platform on windows as well? If yes, what CPU are you on? Cause my CPU even spikes at around 100% when building docker images!

Also I have manually configured to use only 2 cores out 8, but using a browser on windows and then building images in background sure does cause a hell lot of stress on the system.

**Reply by pawanw17** [reply, marked as answer]
I don't use windows much, I have a dual booted machine with ubuntu, and yeah building docker images does take some cpu power(gets the fans running).... If possible just use ubuntu(or any linux) for development, it'll save you from a lot of headaches.

I don't have any experience with wsl2.

**Reply by pariaspe** [reply]
I'm sorry guys, but I can't help you here. No clues about it

**Reply by Bhardwaj-Himanshu** [reply]
Yep, finally switched to ubuntu! and the transition was smooth I must say!
Also thanks for the support!
